BIDVerifyDocument.js

```diff
@@ -94,6 +98,142 @@ const verifyDocument = async (dvcId, verifications, document) => {
+    const tenantInfo = BIDSDK.getTenant();
+
+    let sharedKey = BIDECDSA.createSharedKey(keySet.prKey, docVerifyPublicKey);
+    let userUIDAndDid = uuidv4();
+
+    const encryptedRequestId = BIDECDSA.encrypt(JSON.stringify({
+      ts: Math.round(new Date().getTime() / 1000),
+      appid: 'fixme',
+      uuid: uuidv4()
+    }), sharedKey);
+
+    const headers = {
+      'Content-Type': 'application/json',
+      charset: 'utf-8',
+      publickey: keySet.pKey,
+      licensekey: BIDECDSA.encrypt(licenseKey, sharedKey),
+      requestid: encryptedRequestId
+    };
+
+    const req = {
+      dvcID: dvcId,
+      sessionRequest: {
+        tenantDNS: tenantInfo.dns,
+        communityName: communityInfo.community.name,
+        documentType,
+        userUID: userUIDAndDid,
+        did: userUIDAndDid
+      }
+    };
+
+    const encryptedData = {
+      data: BIDECDSA.encrypt(JSON.stringify(req), sharedKey)
+    };
+
+    let api_response = await fetch(sd.docuverify + "/document_share_session/create", {
+      method: 'post',
+      body: JSON.stringify(encryptedData),
+      headers: headers
+    });
+
+    let status = api_response.status;
+
+    if (api_response) {
+      api_response = await api_response.json();
+    }
+
+    if (status !== 200) {
+      return {
+        error_code: api_response.code,
+        message: api_response.message
+      }
+    }
+
+    return ({ sessionId: api_response.sessionId, sessionUrl: api_response.url });
+
+  } catch (error) {
+    throw error;
+  }
+}
+
+const pollSessionResult = async (dvcId, sessionId) => {
+  try {
+    const keySet = BIDSDK.getKeySet();
+    const licenseKey = BIDSDK.getLicense();
+    const sd = await BIDSDK.getSD();
+    const docVerifyPublicKey = await getDocVerifyPublicKey();
+
+    let sharedKey = BIDECDSA.createSharedKey(keySet.prKey, docVerifyPublicKey);
+
+    const encryptedRequestId = BIDECDSA.encrypt(JSON.stringify({
+      ts: Math.round(new Date().getTime() / 1000),
+      appid: 'fixme',
+      uuid: uuidv4()
+    }), sharedKey);
+
+    const headers = {
+      'Content-Type': 'application/json',
+      charset: 'utf-8',
+      publickey: keySet.pKey,
+      licensekey: BIDECDSA.encrypt(licenseKey, sharedKey),
+      requestid: encryptedRequestId
+    };
+
+    const req = {
+      dvcID: dvcId,
+      sessionId
+    };
+
+    const encryptedData = {
+      data: BIDECDSA.encrypt(JSON.stringify(req), sharedKey)
+    };
+
+    let api_response = await fetch(sd.docuverify + "/document_share_session/result", {
+      method: 'post',
+      body: JSON.stringify(encryptedData),
+      headers: headers
+    });
+
+    let status = api_response.status;
+
+    if (api_response) {
+      api_response = await api_response.json();
+
+      if (api_response.data) {
+        let dec_data = BIDECDSA.decrypt(api_response.data, sharedKey);
+        api_response = JSON.parse(dec_data);
+      }
+    }
+
+    if (status !== 200) {
```

Comment:
I don; think this manipulation should be done here. 
docuverify is already programmed to work properly.. return status/response as-is.

Reply:
Did you check if you need any "decryption" here.

Reply:
i verified that we dont need decryption in error cases